lambda/matching.ml

```diff
@@ -3705,50 +3705,50 @@ and do_compile_matching ~scopes repr partial ctx pmh =
       let ph = what_is_cases pm.cases in
       let pomega = Patterns.Head.to_omega_pattern ph in
       let ploc = head_loc ~scopes ph in
+      let compile_no_test divide combine =
+        compile_no_test ~scopes divide combine repr partial ctx pm
+      in
+      let compile_test divide combine =
+        compile_test
+          (compile_match ~scopes repr partial)
+          partial divide combine ctx pm
+      in
       let open Patterns.Head in
```

Comment:
```suggestion
```

This local `open` isn't needed here and made the change slightly harder to review.

Reply:
I think that the person who wrote this `open` (probably me?) wanted to avoid relying on type-based disambiguation of constructors that are not in the lexical scope, a feature I still have mixed feelings about. I'll go with the easiest option of changing nothing at all here.